core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -88,6 +91,80 @@ public int getValue() {
   }
 
   /**
+   * A class used to cache UnderFileSystems.
+   */
+  private static class Cache {
+    private ConcurrentHashMap<Key, UnderFileSystem> mUnderFileSystemMap = new ConcurrentHashMap<>();
```

Comment:
also please javadoc this variable e.g., what is the key and what is the mapped value.